ui/utils/custom-search.js

```diff
@@ -55,16 +55,14 @@ const SearchBar = ({ onSearch, placeholder, expanded, setExpanded, value = '' })
   const searchRef = useRef(null);
   const classes = useStyles();
 
-  const debouncedOnSearch = debounce(onSearch, 500);
-
+  const debouncedValue = useDebounce(searchText, 500);
```

Comment:
Thankyou @senthil-k8s for working on it, we can use the lodash debounce function here instead of defining our own hook, otherwise, it will add unnecessary extra lines of code to manage.

Reply:
@Yashsharma1911 , using lodash debounce() leads to the same issue, calling multiple times debounce() in SeachBar, and it's parent component. And also we cannot able to override the debounce() of lodash.

Solution is we can improve the existing debounce helper function by making it as a custom hook, 
```jsx
export function useDebounce(func, delay=500) {
  const timer = useRef(null);

  const debouncedFunction = (...args) => {
    clearTimeout(timer.current);
    timer.current = setTimeout(() => {
      func(...args);
    }, delay);
  };

  useEffect(() => {
    return () => {
      clearTimeout(timer.current);
    };
  }, []);

  return debouncedFunction;
}
```

Else the code provided by myself

```jsx
import { useState, useEffect } from 'react';

export function useDebounce(value, delay = 300) {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);
    return () => clearTimeout(timeout);
  }, [value, delay]);
  return debouncedValue;
}


@aabidsofi19 , @Yashsharma1911 . Compare the above the two hooks, which one suits for our codebase. Writing our custom hook, can leads to avoid repetitive call of debounce() in multiple places. 
```

Reply:
By the way adding a custom hook resolves problem everywhere without need of changing multiple files. 


https://github.com/meshery/meshery/assets/93466587/c1e84c03-7ba3-40b6-90db-8cc5ea3017c0

Reply:
we dont want to write our own debouncing mechanism , using  _.debounce with useCallback is enough

Reply:
thanks @aabidsofi19 , working as expected